client/client-multi/client-cpp/kaa/NetworkAddrChangeDetector/NetworkAddrChangeDetector.hpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2014-2016 CyberVision, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef _NETWORK_ADDR_CHANGE_DETECTOR_
+#define _NETWORK_ADDR_CHANGE_DETECTOR_
+
+#include <string>
+#include <stdexcept>
+#include <functional>
+
+namespace kaa {
+
+/**
+ * The class is responsible for observation ip address changes and notifying of the subscriber
+ * if the ip address has been changed.
+ */
+class NetworkAddrChangeDetector {
```

Comment:
The class _does_ the check. It is important because It is wrong to notify user if  IP address hasn't been changed.

> Furthermore, not checking currentIp != ip allows more usage by the user (e.g., check if we reconnect at all).

I see no benefits. The feature is not about notifying the user about reconnections, it is about IP address changes notifications. There is _failover feature_ which is responsible for _reconnections notifications_. 

On my opinion the check should be performed. Probably a separate class for such simple logic isn't the best solution, but the check should be somewhere and moving it to the _KaaClientContext_ isn't the best solution too cuz I see no connection between client context and ip address change detection feature which itself related to the Client (Endpoint). Let's discuss it.